pkg/apis/azure/cache/v1alpha1/redis_types.go

```diff
@@ -0,0 +1,244 @@
+}
+
+// RedisStatus defines the observed state of Redis
+type RedisStatus struct {
+	corev1alpha1.ConditionedStatus
+	corev1alpha1.BindingStatusPhase
+	State   string `json:"state,omitempty"`
+	Message string `json:"message,omitempty"`
+
+	// Endpoint of the Redis resource used in connection strings.
+	Endpoint string `json:"endpoint,omitempty"`
+
+	// Port at which the Redis endpoint is listening.
+	Port int `json:"port,omitempty"`
+
+	// SSLPort at which the Redis endpoint is listening.
+	SSLPort int `json:"sslPort,omitempty"`
+
+	// RedisVersion the Redis endpoint is running.
+	RedisVersion string `json:"redisVersion,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Redis is the Schema for the instances API
+// +k8s:openapi-gen=true
+// +groupName=cache.azure
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.state"
+// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
+// +kubebuilder:printcolumn:name="VERSION",type="string",JSONPath=".status.redisVersion"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+type Redis struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RedisSpec   `json:"spec,omitempty"`
+	Status RedisStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RedisList contains a list of Redis
+type RedisList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Redis `json:"items"`
+}
+
+// NewRedisSpec creates a new RedisSpec
+// from the given properties map.
+func NewRedisSpec(properties map[string]string) *RedisSpec {
+	spec := &RedisSpec{
+		ReclaimPolicy: corev1alpha1.ReclaimRetain,
+
+		// Note that these keys should match the JSON tags of their respective
+		// RedisSpec fields.
+		ResourceGroupName:  properties["resourceGroupName"],
+		Location:           properties["location"],
+		StaticIP:           properties["staticIP"],
+		SubnetID:           properties["subnetId"],
+		RedisConfiguration: parseMap(properties["redisConfiguration"]),
+
+		// TODO(negz): Remove the sku prefix here? This feels clearer, but the
+		// Azure SQL server equivalent uses bare "tier", "vcores", etc.
+		SKU: SKUSpec{
+			Name:   properties["skuName"],
+			Family: properties["skuFamily"],
+		},
+	}
+
+	if b, err := strconv.ParseBool(properties["enableNonSslPort"]); err == nil {
+		spec.EnableNonSSLPort = b
+	}
+	if i, err := strconv.Atoi(properties["shardCount"]); err == nil {
+		spec.ShardCount = i
+	}
+	if i, err := strconv.Atoi(properties["skuCapacity"]); err == nil {
+		spec.SKU.Capacity = i
+	}
+
+	return spec
+}
+
+// parseMap parses a string of comma separated key value pairs, for example
+// "key1: value1, key2: value2", into a string map.
+func parseMap(s string) map[string]string {
```

Comment:
would this `parseMap` func be useful in the util package also?

Reply:
See my comment on `parseSlice`.